rubicon_ml/client/mixin.py

```diff
@@ -198,15 +199,59 @@ def artifacts(self):
         """
         project_name, experiment_id = self._get_parent_identifiers()
 
-        self._artifacts = [
-            client.Artifact(a, self)
-            for a in self.repository.get_artifacts_metadata(
-                project_name, experiment_id=experiment_id
-            )
-        ]
+        if name is not None:
+            self._artifacts = [
+                client.Artifact(a, self)
+                for a in self.repository.get_artifacts_metadata(
+                    project_name, experiment_id=experiment_id
+                )
+                if a.name == name
+            ]
+        else:
+            self._artifacts = [
+                client.Artifact(a, self)
+                for a in self.repository.get_artifacts_metadata(
+                    project_name, experiment_id=experiment_id
+                )
+            ]
 
         return self._artifacts
 
+    def artifact(self, name=None, id=None):
+        """Get an artifact logged to this project by id or name.
+
+        Parameters
+        ----------
+        id : str
+            The id of the artifact to get.
+        name : str
+            The name of the artifact to get.
+
+        Returns
+        -------
+        rubicon.client.Artifact
+            The artifact logged to this project with id `id` or name 'name'.
+        """
+        if (name is None and id is None) or (name is not None and id is not None):
+            raise ValueError("`name` OR `id` required.")
+
+        if name is not None:
+            artifacts = [a for a in self.artifacts() if a.name == name]
+            if len(artifacts) == 0:
+                raise RubiconException(f"No artifact found with name {name}.")
```

Comment:
you could move the exception into the `artifacts` function then use `self.artifacts(name=name)` here instead directly of just `self.artifacts()`. then the exception would be thrown from both functions and not be duplicated